### PR TITLE
Added an experimental flag to trigger JSON flow of go-storage-client

### PR DESCRIPTION
**Pull request by raj-prince**
### Description

- By default, GCSFuse uses xml endpoint for read operations and json endpoint for remaining operations.
- Added a flag, `--experimental-json-read` to trigger the json flow for read operations. 

#### Why is it required?
Both `xml` & `json` flow uses different url-path.  If any user passes endpoint of fake-gcs-server via `--endpoint` flag and that server doesn't support XML path, it starts throwing `not found error`.  In such cases, this flag will be handy to use JSON endpoint for read too. 

Keeping this flag experimental, since we haven't done thorough performance testing.

* Read call without `--experimental-json-read` flag: **/test-bucket/c.txt**
```
127.0.0.1 - - [10/Aug/2023:04:18:34 +0000] "GET /test-bucket/c.txt?generation=1691640877263 HTTP/1.1" 206 7 "-" "gcsfuse/unknown (Go version go1.22-20230729-RC00 cl/552016856 +457721cd52 X:fieldtrack,boringcrypto) (GPN:gcsfuse)"
```

* Read call with `--experimental-json-read` flag: **/storage/v1/b/test-bucket/o/c.txt**
```
127.0.0.1 - - [10/Aug/2023:04:20:41 +0000] "GET /storage/v1/b/test-bucket/o/c.txt?alt=media&generation=1691641162494&prettyPrint=false&projection=full HTTP/1.1" 206 7 "-" "gcsfuse/unknown (Go version go1.22-20230729-RC00 cl/552016856 +457721cd52 X:fieldtrack,boringcrypto) (GPN:gcsfuse)"
```

### Link to the issue in case of a bug fix.
NA

### Testing details
1. Manual tested all the basic operations with and without --experiment-enable-json-read flag, working fine.
3. Unit tests - automation.
4. Integration tests - Run locally, passed successfully.

### CODE REVIEW

File: flags.go [hunk 1]
```diff
@@ -296,6 +296,11 @@ func newApp() (app *cli.App) {
 				Usage: "The format of the log file: 'text' or 'json'.",
 			},
 
+			cli.BoolFlag{
+				Name:  "experimental-json-read",
```
Comment (vadlakondaswetha) [inline flags.go:300]: experimental-enable-json-read

File: internal/storage/storage_handle_test.go [hunk 2]
```diff
@@ -109,3 +109,10 @@ func (t *StorageHandleTest) TestNewStorageHandleWhenUserAgentIsSet() {
 
 	t.invokeAndVerifyStorageHandle(sc)
 }
+
+func (t *StorageHandleTest) TestNewStorageHandleWhenJsonReadEnabled() {
+	sc := storageutil.GetDefaultStorageClientConfig()
+	sc.ExperimentJsonRead = true
+
+	t.invokeAndVerifyStorageHandle(sc)
```
Comment (vadlakondaswetha) [inline internal/storage/storage_handle_test.go:117]: can't we trigger an actual read in bucket)_handle unit test and see if json reads are working?

File: flags.go [hunk 3]
```diff
@@ -395,6 +400,7 @@ type flagStorage struct {
 	OtelCollectorAddress      string
 	LogFile                   string
 	LogFormat                 string
+	ExperimentalJsonRead      bool
```
Comment (vadlakondaswetha) [inline flags.go:403]: you need to add this flag in tools/mount_gcsfuse/main.go

**Review by vadlakondaswetha — CHANGES_REQUESTED** [review]

File: internal/storage/storage_handle_test.go (hunk 2 above)
Comment (raj-prince) [reply internal/storage/storage_handle_test.go:117]: We can't test the json read workflow as part of unit test. As client creation part is hardcoded right now and I don't find any option to pass the option WithJsonReads in the fakestorage package.

See here - https://github.com/fsouza/fake-gcs-server/blob/09b807c1d1c02e277d61c48a6ec99e730b7cdf47/fakestorage/server.go#L479

We can test it via integration test, but hard to check, if it's happening with xml/json programmatically. I have added the argument in our codebase to check basic operation.

File: flags.go [hunk 4]
```diff
@@ -392,11 +397,12 @@ type flagStorage struct {
...
-	StackdriverExportInterval time.Duration
-	OtelCollectorAddress      string
-	LogFile                   string
-	LogFormat                 string
-	DebugFuseErrors           bool
+	StackdriverExportInterval   time.Duration
+	OtelCollectorAddress        string
+	LogFile                     string
+	LogFormat                   string
+	ExperimentalEnableJasonRead bool
```
Comment (vadlakondaswetha) [inline flags.go:404]: typo : jason -> json

**Review by vadlakondaswetha — APPROVED** [review]

**Review by sethiay — APPROVED** [review]

File: tools/integration_tests/operations/operations_test.go [hunk 5]
```diff
@@ -87,7 +87,8 @@ func TestMain(m *testing.M) {
 	setup.ParseSetUpFlags()
 
 	flags := [][]string{{"--implicit-dirs=true"},
-		{"--implicit-dirs=false"}}
+		{"--implicit-dirs=false"},
```
Comment (sethiay) [inline tools/integration_tests/operations/operations_test.go:90]: please add the command for this flag combination to https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/tools/integration_tests/run_tests_mounted_directory.sh

**Pull request merged**